pkg/autodiscovery/listeners/docker.go

```diff
@@ -146,58 +144,7 @@ func (l *DockerListener) init() {
 	}
 
 	for _, co := range containersList {
-		if l.isExcluded(ctx, co) {
-			continue // helper method already logs
-		}
-		var svc Service
-
-		checkNames, err := getCheckNamesFromLabels(co.Labels)
-		if err != nil {
-			log.Errorf("Error getting check names from docker labels on container %s: %v", co.ID, err)
-		}
-
-		containerImage, err := l.dockerUtil.ResolveImageName(ctx, co.Image)
-		if err != nil {
-			log.Warnf("Error while resolving image name: %s", err)
-			containerImage = ""
-		}
-		metricsExcluded := false
-		logsExcluded := false
-		for _, name := range co.Names {
-			if l.filters.IsExcluded(containers.MetricsFilter, name, containerImage, "") {
-				metricsExcluded = true
-			}
-			if l.filters.IsExcluded(containers.LogsFilter, name, containerImage, "") {
-				logsExcluded = true
-			}
-			if metricsExcluded && logsExcluded {
-				break
-			}
-		}
-
-		if findKubernetesInLabels(co.Labels) {
-			svc = &DockerKubeletService{
-				DockerService: DockerService{
-					cID:           co.ID,
-					adIdentifiers: l.getConfigIDFromPs(ctx, co),
-					checkNames:    checkNames,
-					// Host and Ports will be looked up when needed
-				},
-			}
-		} else {
-			svc = &DockerService{
-				cID:             co.ID,
-				adIdentifiers:   l.getConfigIDFromPs(ctx, co),
-				hosts:           l.getHostsFromPs(co),
-				ports:           l.getPortsFromPs(co),
-				creationTime:    integration.Before,
-				checkNames:      checkNames,
-				metricsExcluded: metricsExcluded,
-				logsExcluded:    logsExcluded,
-			}
-		}
-		l.newService <- svc
-		l.services[co.ID] = svc
+		l.createService(ctx, co.ID)
```

Comment:
since we already have the `container`, why not sending it directly to `createService()` instead of just sending the ID? 
Is it to re-use the existing  `createService` function? 

If yes, maybe `createService` can be split in two:
* createServiceFromContainerID
* createServiceFromContainer

Reply:
We can't, `RawContainerList` returns a different data type from `Inspect`, and doesn't have all the information we need (iirc it was the time when the container stopped).